package com.elienai.springfood.domain.exception;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertSame;

import org.junit.jupiter.api.Test;

public class NegocioExceptionTest {

	@Test
	void testNegocioException_comMensagem() {
		NegocioException ex = new NegocioException("Erro de neg贸cio");
		
		assertEquals("Erro de neg贸cio", ex.getMessage());
		assertNull(ex.getCause());
	}
	
	@Test
	void testNegocioException_comMensagemECausa() {
		Throwable causa = new RuntimeException();
		NegocioException ex = new NegocioException("Erro de neg贸cio", causa);
		
		assertEquals("Erro de neg贸cio", ex.getMessage());
		assertNotNull(ex.getCause());
		assertSame(causa, ex.getCause());
	}
	
}
